File: 0367-valid-perfect-square/0367-valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {

        if(num==1){
            return true;
        }

        int left = 0;
        int right = num;

        while(left<=right){
            
            int mid = left + (right-left)/2;

            double square = (double)mid * mid;

            if(square==num){
                return true;
            }

            if(square<num){
                left = mid+1;
            }else{
                right = mid-1;
            }

        }

        return false;
        
    }
};

File: 0367-valid-perfect-square/0367-valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        
        if num == 1:
            return True

        left, right = 0, num

        while left<=right:
            mid = left + (right-left)//2

            if mid*mid == num:
                return True
            
            if mid*mid<num:
                left = mid + 1
            else:
                right = mid - 1

        return False